site/using-smart-accounts/estimate-gas/estimate-user-op-gas-state-overrides.md

```diff
@@ -0,0 +1,121 @@
+  - - meta
+    - name: description
+      content: Follow this guide to learn how to estimate gas of a user operation involving state overrides using Account Kit, a vertically integrated stack for building apps that support ERC-4337 and ERC-6900.
+  - - meta
+    - property: og:description
+      content: Follow this guide to learn how to estimate gas of a user operation involving state overrides using Account Kit, a vertically integrated stack for building apps that support ERC-4337 and ERC-6900.
+  - - meta
+    - name: twitter:title
+      content: How to estimate gas of a user operation involving state overrides
+  - - meta
+    - name: twitter:description
+      content: Follow this guide to learn how to estimate gas of a user operation involving state overrides using Account Kit, a vertically integrated stack for building apps that support ERC-4337 and ERC-6900.
+---
+
+# How to estimate gas of a user operation involving state overrides
+
+## [`StateOverride`](/resources/types#StateOverride)
+
+State overrides allow you to customize the network state for the purpose of your simulation. This feature is useful when you need to simulate or estimate gas for transaction scenarios under conditions that are not currently present on the live network.
+
+To illustrate how state overrides work, let’s use the DAI stablecoin contract. In the DAI system, certain actions, like minting new DAI tokens, can only be performed by addresses designated as "wards". If your address is not a ward, you cannot test minting operations.
+
+### Setting up as a ward through state overrides
+
+To estimate gas for the user operation transaction, you need to simulate a transaction where you mint DAI. This requires overriding the state to make our address appear as a ward. In other words, you need to pass in the state overrides value, along with the user operation transaction data, to adjust the wards map in the DAI contract to include your address with the necessary permissions.
+
+The state override is a mapping of specific storage locations in the smart contract as keys of the map, pointing to the override values for the simulation.
+
+```ts
+{
+  stateOverrides: {
+    "0xDAI_CONTRACT_ADDRESS": {
+      storage: {
+        "0xSTORAGE_SLOT_OF_YOUR_ADDRESS": "0x0000000000000000000000000000000000000000000000000000000000000001"
+      }
+    }
+  }
+}
+```
+
+::: tip Storage Slot Calculation
+Calculating the storage location depends on the structure of the contract’s storage and the specific variable you want to override. For the DAI contract example, the storage location for the wards mapping for your address is calculated based on the [Solidity storage layout rules](https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#mappings-and-dynamic-arrays):
+
+```ts
+const yourAddress = "0x...";
+const overrideStorageLocation = keccak256(
+  concatenate(hexZeroPad(yourAddress), hexZeroPad(0x0))
+); // 0xSTORAGE_SLOT_OF_YOUR_ADDRESS
```

Comment:
After computing `overrideStorageLocation`, we might want to provide further examples of how to use that to pass state overrides to `estimateUserOperationGas`. This page never actually explains what parameter to pass to use the state override.

It's probably enough to have an example, and link to Geth's docs if people want more information about the format of the state overrides.

Reply:
good catch. I forgot to add the example. Added!